codegen/gql_code_builder/lib/src/required_vars_constructor.dart

```diff
@@ -37,9 +37,16 @@ Constructor builtCreateConstructor({
   final assignments = filteredGetters.map((g) {
     final typeDefinitionNode = getTypeDefinitionNode(
         schemaSource.document, g.returns!.symbol!.replaceFirst("G", ""));
-    final isBuiltType = typeDefinitionNode is InputObjectTypeDefinitionNode ||
-        typeDefinitionNode is ScalarTypeDefinitionNode;
 
+    /// "$BuiltList" is a String "BuiltList<dynamic>" and
+    /// [g.returns!.symbol!] returns "BuiltList"
+    ///  so we cannot use equality operator here.
+    final isBuiltList = g.returns?.symbol != null
```

Comment:
can you also check the URL of the reference is from built_value, so we cannot have false positives here?

Reply:
Good call, I thought of doing that too to find `Built` Types at one point (when I wrote some of that code), but I think I didn't end up doing that as it was extremely hard to pin down the correct import. 
For `BuiltList` s though, this would be checking for the built_lists import though, which doesn't change as often, right?


An alternative that might be worth exploring though, would be to check if all `BuiltList` instances are of type `ListType` and `ListType` is exclusively used for `BuiltLists`. Then you could "simply" add another check to L47

Reply:
I think the `url` of the `Reference` should be '"package:built_collection/built_collection.dart", I think this could just be added additionally to this check to avoid clashes with types from the schema, which theoretically also could contain symbols that contain the name `BuiltList`

Reply:
You meant `built_collection` right? Done ✅